pkg/controllers/endpointaccessible/endpoint_accessible_controller.go

```diff
@@ -55,6 +55,12 @@ func (c *endpointAccessibleController) sync(ctx context.Context, syncCtx factory
 		return err
 	}
 
+	// the endpoint list function can return empty results without error when the endpoints does not exists
+	// requeue in that case and wait until we get valid endpoints to check.
+	if len(endpoints) == 0 {
+		return factory.SyntheticRequeueError
```

Comment:
if this is the case, we should be degraded.  This shouldn't be a synthentic error.

Reply:
> if this is the case, we should be degraded. This shouldn't be a synthentic error.

actuall, set the AVailable to false.

Reply:
@deads2k updated, we go available=false in case no endpoints are returned.